File: pbinfo_probs/ultimeleimpare.cpp
#include <iostream>

using namespace std;
int n, i, a[10001], x1=2, x2=2;
int main()
{
    cin>>n;
    for(i=1;i<=n;i++)
        cin>>a[i];
    for(i=n;i>=1;i--)
    {
        if(a[i]%2!=0)
        {
            x2=a[i];
            a[i]=2;
            break;
        }
    }
    for(i=n;i>=1;i--)
    {
        if(a[i]%2!=0)
        {
            x1=a[i];
            a[i]=2;
            break;
        }
    }
    if(x1%2==0 || x2%2==0)
        cout<<"Numere insuficiente";
    else
        cout<<x1<<" "<<x2;
    return 0;
}

File: pbinfo_probs/ultimeleimpare.py
import sys
goods=[]
final=[]
k=0
a=0; b=0

s=sys.stdin.readlines()
for word in s:
	goods.append(word)



goods=[i.replace('\n','') for i in goods]

for i in range(0, len(goods)):
	goods[i]=goods[i].split()

for lst in goods:
	for i in lst:
		final.append(i)



for nr in final[:0:-1]:
	if int(nr) % 2 != 0:
		k+=1
		if k > 2:
			break
		if k == 1:
			b=int(nr)
		else:
			a=int(nr)

if a==0 or b==0:
	print "Numere insuficiente"
else:
	print a, b